### DISCUSSION TITLE
console.log() appear in both server (terminal) and client (browser), why ?

### DISCUSSION THREAD

**Discussion by indrazm** [vercel/next.js#59329] [App Router]
Hey,

I found something weird.
When i use 14.0.+

When i do console.log() in client component, it's appear on the terminal and client.
I believe i remember it was only appear in browser without server.

Is it normal ? Or is there any behaviour change ?

**Comment by icyJoseph** [comment]
Client components, are the components we've used up till now, and just like that, they are SSR'd:

- https://github.com/reactwg/server-components/discussions/4

That is, they do run in the server too, to collect their initial HTML, and what not.

The different thing is that Server Components only render server side, so the logs created in those, would not show in the browser.

**Reply by oxillix** [reply]
How to disable the logging in terminal? It's confusing me a bit, lots of undefined objects, and is polluting my ssr terminal

**Comment by Julian24Design** [comment]
I'm using 14.0.2 and the console.log in client components only works in the browser.

**Comment by KenSoftDev** [comment]
I am also experiencing confusion as the logs from the Client Component are appearing both in the browser console and the development server console. I am using Next.js 14.2.3.

**Reply by icyJoseph** [reply]
Client components are also SSR'd... They are the components we have used so far. The new thing are server components, which render only in the server. 

- https://github.com/reactwg/server-components/discussions/4

**Comment by philipp-serfling** [comment]
So we have an explanation for the behavior but is there also a workaround?
